packages/react-performance/README.md

```diff
@@ -17,7 +17,36 @@ $ yarn add @shopify/react-performance
 
 The most basic way to use the tools in this package is to record information and display it locally to the user. In practice you usually only want to do this in development so that developers can easily see performance information.
 
-#### Display data using NavigationListener
+### Adding some `<PerformanceMark />`s
+
+We will need to add some `<PerformanceMark />` components to our React application in order to help `@shopify/react-performance` determine when your application is fully loaded and React has mounted your components.
+
+```tsx
+//Cats.tsx
+import React from 'react';
+import {PerformanceMark} from '@shopify/react-performance';
+
+/**
+ * Represents an entire page of our application
+ */
+export function Cats() {
+  return (
+    <section>
+      <h1>Cats</h1>
+      <ul>
+        <li>Kokusho</li>
+        <li>Moe</li>
+        <li>Smoky</li>
+      </ul>
+      <PerformanceMark stage="complete" id="cats" />
+    </section>
+  );
+}
+```
+
+In production apps, it is important to always have one `<PerformanceMark stage="complete" />` on each page level component. Otherwise, client-side navigations may not be recorded properly.
```

Comment:
```suggestion
In production apps, it is important to always have one `<PerformanceMark stage="complete" />` on each page level component. Otherwise, client-side navigations may not be recorded properly. You should not use `usePerformanceMark("complete", "id")` in place of the component because...
```

Reply:
The `...` is mostly in reference to a question I asked in a previous PR on why we dont use `usePerformanceMark` , I think the reason why is a hook version will get involved before the entire component finished rendering?

Reply:
I'll add a tiny bit more context here, but it feels more like going super deep on when and where you might miss performance marks and the ramifications belongs more in an "FAQ" or something.

Reply:
Also
> Where does Web include performance marks that aren't just in the pages?

From what I can see it literally *only* uses `usable` and `complete` (mostly `complete`, and our types currently disallow custom stages) and  I *think* only on page level components.

Reply:
https://github.com/Shopify/web/blob/master/app/foundation/ErrorBoundary/ErrorBoundary.tsx#L55 and probably should typically put one in the `NotFound` component to be safe

Reply:
so I didnt add this in online-store-web yet but I do kind of wonder. Would this also make sense to add for an embedded modal? or just a large modal?